### PR TITLE
[DOC] Restructure & Cleaner API reference to manually curate public APIs

### LINKED ISSUE #1935: [ENH] cleaner API reference
The API reference is automatically generated and somewhat confusing now - the overview includes private modules and classes that are not useful for the typical user.

Instead, I would recommend to remodel the page based on `sktime`, and manually populate the API reference, avoiding to list private classes.
See sktime API reference: https://github.com/sktime/sktime/tree/main/docs/source/api_reference

I would also suggest that we:

* keep the split in `models`, `metrics`, `data` etc
* on individual pages, we have the ordering:
    * at the top, a short preamble explaining the different types of objects
    * first, neural networks
    * second, package containers
    * third, v2 objects, clearly marking them as beta

### PR DESCRIPTION
**Pull request by PredictiveManish**
#### Reference Issues/PRs
Closes #1935 
Relates to improving API documentation discoverability and user experience.

#### What does this implement/fix? Explain your changes.

This PR restructures the API reference documentation from auto-generated content to manually curated public APIs, following the model used by sktime.

**Changes:**
- Replaced auto-generated recursive API documentation with manually maintained `.rst` files for each module category
- Created four new API reference files:
  - `docs/source/api/data.rst` - Data loading and preprocessing utilities
  - `docs/source/api/models.rst` - Neural network architectures
  - `docs/source/api/metrics.rst` - Loss functions and evaluation metrics
  - `docs/source/api/utils.rst` - Utility functions

- Updated `docs/source/api.rst` to use `toctree` directive instead of `autosummary:recursive`
- Organized each module page with consistent structure:
  - Introductory preamble explaining module purpose
  - Neural Networks section (primary implementations)
  - Package Containers section (wrapper/utility classes)
  - V2 Objects section (clearly marked as beta/experimental)

**Benefits:**
- Eliminates private modules and classes from API documentation
- Improves discoverability for typical users
- Provides better organization and navigation
- Simplifies the addition of custom sections and examples

#### What should a reviewer concentrate their feedback on?

- Completeness of public API coverage across all four modules
- Accuracy of class/function categorization (neural networks vs. containers vs. utilities)
- Clarity of the introductory preambles for each module
- Proper identification and marking of v2/beta objects
- Removal of any private APIs that shouldn't be documented

#### Did you add any tests for the change?

Not applicable - this is a documentation restructuring change.

#### Any other comments?

The actual public API items in the autosummary directives should be verified against the current codebase to ensure completeness and accuracy.

#### PR checklist

- [x] The PR title starts with [DOC]
- [ ] Added/modified tests (N/A - documentation only)
- [x] Used pre-commit hooks when committing

### CODE REVIEW

**Comment by PredictiveManish** [comment]
@phoeenniixx I think this will work! Please let me know what else improvement it requires

File: docs/source/tutorials/stallion.ipynb [hunk 1]
```diff
@@ -41,7 +41,6 @@
    "source": [
     "import warnings\n",
     "\n",
-    "\n",
```
Comment (PredictiveManish) [inline docs/source/tutorials/stallion.ipynb]: This is not an intentional change, it come directly when I executed this command for issues in tests previously
```
pre-commit run --all-files
```
